src/components/app/filter-dropdown.jsx

```diff
@@ -26,11 +25,13 @@ const FilterCategory = React.createClass({
       }
       return true;
     });
-  },
-  onFilterInputChange(e) {
+  };
+
+  onFilterInputChange = (e) => {
     this.setState({filterStr: e.currentTarget.value});
-  },
-  renderItem(item) {
+  };
+
+  renderItem = (item) => {
```

Comment:
I don't think it's necessary to convert method other than event listener to class properties, `this` is still this class because it's not called from event listener.

Related https://stackoverflow.com/questions/44423947/javascript-class-methods-versus-properties#comment75847321_44424064

Reply:
I'm not sure, but it seems to me that the function `renderItem` is indeed called from outside. I suspect it is called somewhere in `react-bootstrap` library.

Reply:
1. It literally says `It's actually not necessary to transform all methods to arrow functions`
2. That's right, and we're talking about comparing `bind(this)` and `() => listener()`, other function besides event listener does not need bind.

If we're deciding to move back to using normal class method on non event listener, if that's even worth it, we need to make sure it doesn't break anything and this project has no complete tests.

@andrewda gonna need your opinion on this.

Reply:
Maybe we could leave this as a separate issue? If we decide to move back to normal methods, we need careful reviews. Errors are likely to be missed in this large PR.